autoprotocol/builders.py

```diff
@@ -1738,6 +1746,146 @@ def blowout(self, volume, initial_z, flowrate=None):
+        sometimes not sufficient to overcome the surface tension to pull
+        or push the liquid sufficiently through transfer tip, it will
+        default to "positive_displacement" unless otherwise specified.
+
+        Example Usage:
+        .. code-block:: python
+        example_transports = [
+            LiquidHandle.builders.transport(
+                volume=Unit(1, "uL"),
+                density=None,
+                pump_override_volume=Unit(2, "uL"),
+                flowrate=LiquidHandle.builders.flowrate(
+                    target=Unit(10, "uL/s")
+                ),
+                delay_time=Unit(0.5, "s"),
+                mode_params=LiquidHandle.builders.mode_params(
+                    liquid_class="air",
+                    position_z=LiquidHandle.builders.position_z(
+                        reference="preceding_position"
+                    )
+                )
+            ),
+            LiquidHandle.builders.transport(
+                volume=Unit(1, "uL"),
+                density=None,
+                pump_override_volume=Unit(2, "uL"),
+                flowrate=LiquidHandle.builders.flowrate(
+                    target=Unit(10, "uL/s")
+                ),
+                delay_time=Unit(0.5, "s"),
+                mode_params=LiquidHandle.builders.mode_params(
+                    liquid_class="viscous",
+                    position_z=LiquidHandle.builders.position_z(
+                        reference="preceding_position"
+                    )
+                )
+            )
+        ]
+
+        LiquidHandle.builders.desired_mode(example_transports, None)
+
+        This will return "positive_displacement" based on the "viscous" liquid
+        class. Note that "air" (which defaults to "air_displacement") does not
+        cause a conflict since "air" is often applied in additional transfers
+        of air (such as blowout) for any liquid class, and is disregarded.
+
+        Parameters
+        ----------
+        transports : dict, optional
+            Dictionary of the transport parameters
+        mode : str, optional
+            Mode of dispense type
+
+        Returns
+        -------
+        str
+            Mode of dispense type
+
+        Raises
+        ------
+        TypeError
+            mode is not in str
+        ValueError
+            mode does not contain valid mode value
+        TypeError
+            liquid is not of str
+        ValueError
+            liquid_class is not a valid value
+        ValueError
+            liquid_class did not resolve to a mode
+        ValueError
+            multiple liquid_class exists in one LiquidHandle
+        """
+        # dict for default dispense mode for each liquid_class
+        liquid_class_to_dispense_mode = {
+            "air": "air_displacement",
+            "default": "air_displacement",
+            "viscous": "positive_displacement"
+        }
+        if mode:
+            if not isinstance(mode, str):
```

Comment:
is this check redundant with the below one? there's no way for a non-string thing to be in dispense_modes

Reply:
I guess so? It just catches different error type. I can remove it.